app/services/requirements/file_attachment_upload_checker.rb

```diff
@@ -63,8 +70,23 @@ def issues
   private
 
     def unsupported_type?
-      type = Marcel::MimeType.for(file, name: file.original_filename)
-      ALLOWED_FORMATS.exclude? type
+      (ALLOWED_FORMATS + [ALLOWED_ZIP]).exclude? content_type
+    end
+
+    def invalid_zip?
+      return if content_type != ALLOWED_ZIP
+
+      Zip::File.open(file) do |zf|
+        zf.each do |f|
+          type = Marcel::MimeType.for(f.get_input_stream, name: f.name)
+          return true if ALLOWED_FORMATS.exclude? type
```

Comment:
With regards to the zip suggestion I'd check it's an allowed format and not a zip with a comment to say we don't allow nested archives.

Reply:
👍